File: frontend/js/doctor.js
/* ═══════════════════════════════════════════
   MedCore — doctor.js
   Doctor CRUD — Real API integration
═══════════════════════════════════════════ */

const DoctorAPI = {
  async getAll(specialty = '') {
    const params = specialty ? { specialty } : {};
    return api.get('/doctors', params);
  },

  async getById(id)        { return api.get(`/doctors/${id}`); },

  async create(data)       { return api.post('/doctors', data); },

  async update(id, data)   { return api.put(`/doctors/${id}`, data); },

  async updateStatus(id, status) {
    return api.patch(`/doctors/${id}/status?status=${status}`);
  },

  async delete(id)         { return api.delete(`/doctors/${id}`); },
};

async function loadAndRenderDoctors(specialty = '') {
  // showSectionLoading('sec-doctors');
  try {
    const doctors = await DoctorAPI.getAll(specialty);
    DB.doctors = doctors.map(mapDoctorFromApi);
    renderDoctors(DB.doctors);
  } catch (err) {
    showToast('❌ Lỗi tải danh sách bác sĩ: ' + err.message, 'error');
    renderDoctors(DB.doctors);
  }
}

function mapDoctorFromApi(d) {
  return {
    rawId:  d.id,
    id:     d.doctorCode,
    name:   d.fullName,
    spec:   d.specialty,
    degree: d.degree || '—',
    shift:  d.shift?.label || d.shift || '—',
    status: d.status?.label || d.status || '—',
    phone:  d.phone || '—',
    email:  d.email || '—',
    _raw:   d,
  };
}

async function saveNewDoctor() {
  const name   = document.getElementById('d-name')?.value?.trim();
  const degree = document.getElementById('d-degree')?.value;
  const spec   = document.getElementById('d-spec')?.value;
  const shift  = document.getElementById('d-shift')?.value;
  const email  = document.getElementById('d-email')?.value?.trim();
  const phone  = document.getElementById('d-phone')?.value?.trim();

  if (!name) { showToast('Vui lòng nhập tên bác sĩ!', 'error'); return; }

  const shiftMap = { 'Ca sáng':'CA_SANG','Ca chiều':'CA_CHIEU','Ca trực đêm':'CA_TRUC_DEM' };

  try {
    console.info({ fullName: name, degree, specialty: spec, shift: shiftMap[shift] || 'CA_SANG', email, phone });
    await DoctorAPI.create({ fullName: name, degree, specialty: spec, shift: shiftMap[shift] || 'CA_SANG', email, phone });
    closeModal();
    showToast(`✅ Đã thêm bác sĩ ${name}!`, 'success');
    await loadAndRenderDoctors();
  } catch (err) {
    showToast('❌ ' + err.message, 'error');
  }
}

async function updateDoctor(id) {
  const rawId = DB.doctors.find(d => d.id === id)?.rawId;
  console.info(id);
  console.info(rawId);
  if (!rawId) { showToast('Không tìm thấy ID bệnh nhân!', 'error'); return; }

  const name   = document.getElementById('doc-name')?.value;
  const degree = document.getElementById('doc-degress')?.value;
  const spec   = document.getElementById('doc-spec')?.value;
  const status = document.getElementById('doc-status')?.value;
  console.info(status);

  // Map label back to enum key
  const statusMap = { 'Đang khám':'DANG_KHAM', 'Sẵn sàng':'SAN_SANG', 'Nghỉ':'NGHI' };

  try {
    await DoctorAPI.update(rawId, { fullName: name, degree: degree, status: statusMap[status] || status , specialty:spec});
    closeModal();
    showToast('✅ Đã cập nhật hồ sơ!', 'success');
    await loadAndRenderDoctors();
  } catch (err) {
    showToast('❌ ' + err.message, 'error');
  }
}

async function deleteDoctor(id) {
  if (!confirm(`Xác nhận xóa bệnh nhân ${id}?`)) return;
  const rawId = DB.doctors.find(d => d.id === id)?.rawId;
  if (!rawId) return;

  try {
    await DoctorAPI.delete(rawId);
    showToast(`Đã xóa bác sĩ ${id}`, 'warning');
    await loadAndRenderDoctors();
  } catch (err) {
    showToast('❌ ' + err.message, 'error');
  }
}

async function filterDoctors(spec) {
  await loadAndRenderDoctors(spec);
}

File: backend/src/main/java/com/hospital/system/entity/Doctor.java
package com.hospital.system.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "doctors")
@Data @NoArgsConstructor @AllArgsConstructor @Builder
public class Doctor {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "doctor_code", unique = true, nullable = false, length = 20)
    private String doctorCode;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", unique = true)
    private User user;

    @Column(name = "full_name", nullable = false, length = 200)
    private String fullName;

    @Column(nullable = false, length = 100)
    private String specialty;

    @Column(length = 50)
    private String degree;

    @Enumerated(EnumType.STRING)
    private Shift shift = Shift.CA_SANG;

    @Enumerated(EnumType.STRING)
    private Status status = Status.SAN_SANG;

    @Column(length = 15)
    private String phone;

    @Column(length = 150)
    private String email;

    @Column(name = "room_number", length = 20)
    private String roomNumber;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @PrePersist
    void onCreate() { this.createdAt = LocalDateTime.now(); }

    public enum Shift {
        CA_SANG("Ca sáng"), CA_CHIEU("Ca chiều"), CA_TRUC_DEM("Ca trực đêm");
        private final String label;
        Shift(String label) { this.label = label; }
        public String getLabel() { return label; }
    }
    public enum Status {
        SAN_SANG("Sẵn sàng"), DANG_KHAM("Đang khám"), NGHI("Nghỉ");
        private final String label;
        Status(String label) { this.label = label; }
        public String getLabel() { return label; }
    }
}